conductor_api/src/conductor/admin.rs

```diff
@@ -381,22 +388,70 @@ impl ConductorAdmin for Conductor {
         Ok(())
     }
 
-    fn add_agent(&mut self, new_agent: AgentConfiguration) -> Result<(), HolochainError> {
+    fn add_agent(
+        &mut self,
+        id: String,
+        name: String,
+        holo_remote_key: Option<bool>,
+    ) -> Result<String, HolochainError> {
         let mut new_config = self.config.clone();
-        if new_config.agents.iter().any(|i| i.id == new_agent.id) {
+        if new_config.agents.iter().any(|i| i.id == id) {
             return Err(HolochainError::ErrorGeneric(format!(
                 "Agent with ID '{}' already exists",
-                new_agent.id
+                id
             )));
         }
-        new_config.agents.push(new_agent.clone());
+
+        let (keystore, public_address) = if self.using_dpki() {
```

Comment:
As per https://github.com/holochain/holochain-rust/pull/1199, I think we should omit creating a keystore if `holo_remote_key` is set.

Reply:
Done